#2388 : Auto-exec at import

DuckflipXYZ opened the pull request:
Not tested yet

Missing :
- More details for template filters and parameters ... maybe a collapsable tab for write more infos
- Required rights are a bit random

Test case requirements :
A VIP account

Test case :

The idea is to create various template requiring distinct conditions for being performed. Feel free to test absurd cases for template filters and template parameters, I tried to put as much controls as possible.

Then, import dicom respecting (or not)  the templates requirements, and see if the VIP execution is correctly performed, and if the outputs are accessible in Shanoir UI.

a-ba commented:
Hi
I added the env variables in the qualif instances, though i am not fond of the solution.
```sh
#Shanoir shutdown and continuance hours
SHANOIR_SHUTDOWN_HOUR=2
SHANOIR_CONTINUANCE_HOUR=5
```
As i understand it (the doc line is terse), these delimit the time slot when vip jobs are suspended, which solves a particular case (ensure the jobs are not affected by the nightly maintainance operations).

I would rather prefer have « graceful shutdown  » feature, which would solve a more general case (prevent any job or import to be affect, whenever a shutdown is triggered let it be for nightly maintainance or a version upgrade), and be less cumbersome to use (because it would not require any configuration and planning).

A graceful shutdown would be initiated by the sysadmin, and would trigger the following procedure:
1. stop submitting new vip jobs and stop accepting new imports
2. wait until the current jobs and import are completed
3. shutdown the app

The usual way is to use a separate signal to trigger the graceful shutdown (eg: SIGQUIT instead of SIGTERM). The tricky part in shanoir is the pending tasks are distributed in multiple containers (datasets, imports and maybe others).

michaelkain commented:
Hi @DuckflipXYZ, I hope you are doing fine. It might be great to fix the conflicts please? Thank you! With kind regards, Michael

DuckflipXYZ committed:
#2388: env set-up

# Conflicts:
#	docker-compose/common/entrypoint_common
#	docker-compose/database-migrations/db-changes/datasets/0064_create_execution_template.sql

DuckflipXYZ committed:
#2388: env set-up

# Conflicts:
#	docker-compose/common/entrypoint_common
#	docker-compose/database-migrations/db-changes/datasets/0064_create_execution_template.sql

DuckflipXYZ committed:
#2388 : execution-tempalte implementation

DuckflipXYZ committed:
#2388 : execution-tempalte implementation

# Conflicts:
#	shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/repository/DatasetRepository.java
#	shanoir-ng-front/src/app/app.module.ts

# Conflicts:
#	shanoir-ng-datasets/src/main/java/org/shanoir/ng/vip/executionMonitoring/service/ExecutionMonitoringService.java

DuckflipXYZ committed:
#2388 : check for executions at import modified to a runner at shanori startup

DuckflipXYZ committed:
#2388 : check for executions at import modified to a runner at shanori startup

DuckflipXYZ committed:
#2388 : runner at shanoir startup fixes

DuckflipXYZ committed:
#2388 : runner at shanoir startup fixes

DuckflipXYZ commented on shanoir-ng-datasets/src/main/java/org/shanoir/ng/importer/service/ImporterService.java:
```diff
@@ -184,6 +194,16 @@ public void createAllDatasetAcquisition(ImportJob importJob, Long userId) throws
                     + "] in examination [" + examination.getId() + "]");
             eventService.publishEvent(event);
 
+            if (Objects.nonNull(templateService)) { //For maven test
```
Hi @michaelkain, sounds good to me

michaelkain commented:
Hi @DuckflipXYZ, hi @fragkiadamis,
I hope you are doing great.
@DuckflipXYZ sorry again for this, but do you think you might solve the conflicts please?
@fragkiadamis: I wanted to help Guéwen on his auth-issue he is having on this PR, do you think during my absence you could please have a look?
Thank you and with kind regards, Michael

DuckflipXYZ committed:
#2388 : some fixes

DuckflipXYZ committed:
#2388 : some fixes again

fragkiadamis committed:
chore: remove unused StudyUserRight import in execution-template-list component

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
fix: initialize import_exec_count on first startup if missing

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
chore(datasets): remove debugging logs

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
chore(datasets): fix spacing in comments

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
chore(execution-template): add required label to "Processing type" field

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
fix(execution-template): handle empty pipeline list from VIP gracefully

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
fix: handle null case in miscellaneous_parameter insert query

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
fix: StandardEvaluationContext → SimpleEvaluationContext in both SpEL eval sites

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
fix: Added return after null-guard for importExecParam

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
fix: Removed delete(monitoring) from transient-failure catch block

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
fix: swapped args, wrong retry variable, dead if/else

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
fix: remove unnecessary retry logic in acquisition retrieval loop

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
chore(execution-template): remove unused import for "pipeline"

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>

fragkiadamis committed:
fix(execution-template): change security annotations to require ADMIN role only

Signed-off-by: Adam Fragkiadakis <adamfrag@gmail.com>
